src/include/duckdb/function/scalar_function.hpp

```diff
@@ -48,6 +54,9 @@ class ScalarFunction : public SimpleFunction {
 	// The dependency function (if any)
 	dependency_function_t dependency;
 
+	//! An UDF function to execute
+	udf_function_t udf_function;
```

Comment:
I think this variable should not exist, udf functions should share the code path with regular functions. They already share the signature after all. I would remove this variable and use the `function` variable above.